sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -70,9 +70,16 @@ final class CastExprMeta[INPUT <: CastBase](
       case (FloatType | DoubleType, ByteType | ShortType | IntegerType | LongType) if
           doFloatToIntCheck && !conf.isCastFloatToIntegralTypesEnabled =>
         willNotWorkOnGpu(buildTagMessage(RapidsConf.ENABLE_CAST_FLOAT_TO_INTEGRAL_TYPES))
-      case (dt: DecimalType, _: StringType) if dt.precision > DType.DECIMAL64_MAX_PRECISION =>
-        willNotWorkOnGpu(s"decimal to string with a " +
-            s"precision > ${DType.DECIMAL64_MAX_PRECISION} is not supported yet")
+      case (dt: DecimalType, _: StringType) =>
+        if (!conf.isCastDecimalToStringEnabled) {
+          willNotWorkOnGpu("The GPU does NOT produce exact same string as Spark produces, " +
```

Comment:
```suggestion
          willNotWorkOnGpu("the GPU does NOT produce the exact same string as Spark produces, " +
```

Nit: the capitalization of NOT seems a bit excessive given we have plenty of other "because" reasons that don't also capitalize.